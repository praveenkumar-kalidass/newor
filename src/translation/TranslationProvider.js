import React, { useEffect, useState } from 'react';
import LocalizedStrings from 'react-localization';
import PropTypes from 'prop-types';

import CONSTANT from 'constant';
import TranslationContext from './TranslationContext';
import en from './language/en.json';
import ta from './language/ta.json';

const TranslationProvider = ({ children }) => {
  const [translations] = useState({ en, ta });
  const [language, setLanguage] = useState(CONSTANT.LANGUAGE.ENGLISH);
  const [translation, setTranslation] = useState(new LocalizedStrings({ en }));

  useEffect(() => {
    setTranslation(new LocalizedStrings({
      [language]: translations[language],
    }));
  }, [language]);

  return (
    // eslint-disable-next-line react/jsx-no-constructed-context-values
    <TranslationContext.Provider value={{ translation, setLanguage }}>
      {children}
    </TranslationContext.Provider>
  );
};

TranslationProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TranslationProvider;
