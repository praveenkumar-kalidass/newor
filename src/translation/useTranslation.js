import { useContext } from 'react';

import TranslationContext from './TranslationContext';

const useTranslation = () => {
  const { translation, setLanguage } = useContext(TranslationContext);

  const changeTranslation = (language) => {
    setLanguage(language);
  };

  const translate = (key) => {
    const result = key.split('.').reduce((final, value) => {
      if (final[value]) {
        return final[value];
      }
      return '';
    }, translation);

    return result || key;
  };

  return {
    translation,
    translate,
    changeTranslation,
  };
};

export default useTranslation;
