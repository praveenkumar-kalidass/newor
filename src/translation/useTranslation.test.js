import { act, renderHook } from '@testing-library/react-native';

import TranslationProvider from './TranslationProvider';
import useTranslation from './useTranslation';

jest.mock('./useTranslation', () => {
  const actual = jest.requireActual('./useTranslation');
  return actual;
});

describe('useTranslation', () => {
  it('should match snapshot for default translation', () => {
    const { result } = renderHook(() => useTranslation(), {
      wrapper: TranslationProvider,
    });

    expect(result.current.translation).toMatchSnapshot();
  });

  it('should match snapshot for updated translation', () => {
    const { result } = renderHook(() => useTranslation(), {
      wrapper: TranslationProvider,
    });

    act(() => result.current.changeTranslation('ta'));

    expect(result.current.translation).toMatchSnapshot();
  });

  it('should get translated text', () => {
    const { result } = renderHook(() => useTranslation(), {
      wrapper: TranslationProvider,
    });

    expect(result.current.translate('LOGIN')).toStrictEqual('Login');
    expect(result.current.translate('NAN')).toStrictEqual('NAN');
  });
});
