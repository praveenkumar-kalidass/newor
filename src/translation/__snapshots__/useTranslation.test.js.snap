// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useTranslation should match snapshot for default translation 1`] = `
LocalizedStrings {
  "ADD_ASSET": "Add an Asset",
  "ADD_DEPOSIT": "Add a new Deposit",
  "ALREADY_HAVE_ACCOUNT": "Already have an account?",
  "ASSETS": "Assets",
  "ASSET_ADDED_SUCCESSFULLY": "{asset} - Asset added successfully",
  "CONFIRM_PASSWORD": "Confirm Password",
  "CURRENT": "Current Account",
  "DEPOSIT": "Deposit",
  "DEPOSITORY_NAME": "Depository name",
  "DEPOSIT_TYPE": "Deposit type",
  "EMAIL": "Email",
  "EMAIL_VERIFICATION_FAILURE": "Your email verification has failed. Please try again.",
  "EMAIL_VERIFICATION_SUCCESS": "You have successfully verified your email with Newor. Please proceed with login.",
  "ENTER_DEPOSIT_DETAILS": "Kindly enter all details of {depositType}",
  "ENTER_EMAIL_WITH_NEWOR": "Please enter your email registered with Newor",
  "ERROR": "Error",
  "ERROR_CODE": Object {
    "NEWOR_BAD_REQUEST": "Oops! Something went wrong. Please try again.",
    "NEWOR_CLIENT_NOT_FOUND": "Oops! Please try using authorized application.",
    "NEWOR_EMAIL_NOT_VERIFIED": "Oops! You haven't verified your email yet. Please complete your email verification to continue.",
    "NEWOR_INTERNAL_SERVER_ERROR": "Oops! Something went wrong. Please try again.",
    "NEWOR_INVALID_CREDENTIALS": "Uh Oh! Incorrect login credentials. Please try again.",
    "NEWOR_USER_NOT_FOUND": "Oops! We can't find your credentials. Please try signing up first.",
  },
  "FIRST_NAME": "First name",
  "FIXED_DEPOSIT": "Fixed Deposit",
  "FORGOT_PASSWORD": "Forgot Password?",
  "FORGOT_PASSWORD_SUCCESS": "Password reset initiated!",
  "INITIAL_DEPOSIT": "Initial deposit",
  "INTEREST_RATE": "Rate of Interest",
  "LAST_NAME": "Last name",
  "LOGIN": "Login",
  "MATURITY_DATE": "Maturity date",
  "MOBILE_NUMBER": "Mobile number",
  "NEW_TO_NEWOR": "New to Newor?",
  "NO": "No",
  "PASSWORD": "Password",
  "PICTURE_UPDATED": "Picture updated successfully",
  "PLEASE_RESET_PASSWORD_EMAIL": "We have sent a mail to you to reset your password. Please update your password to login.",
  "PLEASE_VERIFY_EMAIL": "We have sent a link to your email for verification. Please complete the verification process to login.",
  "PRESENT_AMOUNT": "Amount present in the account",
  "PROFILE": "Profile",
  "PUBLIC_PROVIDENT_FUND": "Public Provident Fund",
  "RECURRING_DEPOSIT": "Recurring Deposit",
  "REMEMBER_CREDENTIALS": "Remember your credentials?",
  "RESET_PASSWORD": "Reset Password",
  "RESET_PASSWORD_SUCCESS": "You have successfully reset your password for your email. Please proceed with login",
  "SAME_INITIAL_AMOUNT": "Amount currently present in the account is the initial deposit?",
  "SAVINGS": "Savings Account",
  "SELECT_ASSET": "Select your Asset",
  "SELECT_DEPOSIT_TYPE": "Select Deposit type",
  "SELECT_START_AND_MATURITY_DATE": "Select Start date and Maturity date for {depositType}",
  "SELECT_START_DATE": "Select Start date for {depositType}",
  "SIGNUP": "Signup",
  "SIGNUP_SUCCESS": "Hurray! You have successfully signed up!",
  "START_DATE": "Start date",
  "SUBMIT": "Submit",
  "SUCCESS": "Success",
  "SURE_TO_LOGOUT": "Are you sure you want to logout of Newor?",
  "TOTAL_ASSET_VALUE": "Total Asset value",
  "WELCOME_TO_NEWOR": "Welcome to Newor!",
  "YES": "Yes",
  "_defaultLanguage": "en",
  "_defaultLanguageFirstLevelKeys": Array [
    "LOGIN",
    "SIGNUP",
    "WELCOME_TO_NEWOR",
    "EMAIL",
    "MOBILE_NUMBER",
    "PASSWORD",
    "FIRST_NAME",
    "LAST_NAME",
    "SIGNUP_SUCCESS",
    "ALREADY_HAVE_ACCOUNT",
    "NEW_TO_NEWOR",
    "PLEASE_VERIFY_EMAIL",
    "FORGOT_PASSWORD",
    "SUBMIT",
    "REMEMBER_CREDENTIALS",
    "ENTER_EMAIL_WITH_NEWOR",
    "FORGOT_PASSWORD_SUCCESS",
    "PLEASE_RESET_PASSWORD_EMAIL",
    "SUCCESS",
    "ERROR",
    "EMAIL_VERIFICATION_SUCCESS",
    "EMAIL_VERIFICATION_FAILURE",
    "RESET_PASSWORD",
    "CONFIRM_PASSWORD",
    "RESET_PASSWORD_SUCCESS",
    "SURE_TO_LOGOUT",
    "YES",
    "NO",
    "PICTURE_UPDATED",
    "PROFILE",
    "ASSETS",
    "TOTAL_ASSET_VALUE",
    "ADD_ASSET",
    "SELECT_ASSET",
    "DEPOSIT",
    "SAVINGS",
    "CURRENT",
    "FIXED_DEPOSIT",
    "RECURRING_DEPOSIT",
    "PUBLIC_PROVIDENT_FUND",
    "ADD_DEPOSIT",
    "DEPOSIT_TYPE",
    "SELECT_DEPOSIT_TYPE",
    "START_DATE",
    "MATURITY_DATE",
    "SELECT_START_DATE",
    "SELECT_START_AND_MATURITY_DATE",
    "ENTER_DEPOSIT_DETAILS",
    "INITIAL_DEPOSIT",
    "PRESENT_AMOUNT",
    "INTEREST_RATE",
    "SAME_INITIAL_AMOUNT",
    "DEPOSITORY_NAME",
    "ASSET_ADDED_SUCCESSFULLY",
  ],
  "_interfaceLanguage": "en-US",
  "_language": "en",
  "_opts": Object {
    "customLanguageInterface": [Function],
    "logsEnabled": true,
    "pseudo": false,
    "pseudoMultipleLanguages": false,
  },
  "_props": Object {
    "en": Object {
      "ADD_ASSET": "Add an Asset",
      "ADD_DEPOSIT": "Add a new Deposit",
      "ALREADY_HAVE_ACCOUNT": "Already have an account?",
      "ASSETS": "Assets",
      "ASSET_ADDED_SUCCESSFULLY": "{asset} - Asset added successfully",
      "CONFIRM_PASSWORD": "Confirm Password",
      "CURRENT": "Current Account",
      "DEPOSIT": "Deposit",
      "DEPOSITORY_NAME": "Depository name",
      "DEPOSIT_TYPE": "Deposit type",
      "EMAIL": "Email",
      "EMAIL_VERIFICATION_FAILURE": "Your email verification has failed. Please try again.",
      "EMAIL_VERIFICATION_SUCCESS": "You have successfully verified your email with Newor. Please proceed with login.",
      "ENTER_DEPOSIT_DETAILS": "Kindly enter all details of {depositType}",
      "ENTER_EMAIL_WITH_NEWOR": "Please enter your email registered with Newor",
      "ERROR": "Error",
      "ERROR_CODE": Object {
        "NEWOR_BAD_REQUEST": "Oops! Something went wrong. Please try again.",
        "NEWOR_CLIENT_NOT_FOUND": "Oops! Please try using authorized application.",
        "NEWOR_EMAIL_NOT_VERIFIED": "Oops! You haven't verified your email yet. Please complete your email verification to continue.",
        "NEWOR_INTERNAL_SERVER_ERROR": "Oops! Something went wrong. Please try again.",
        "NEWOR_INVALID_CREDENTIALS": "Uh Oh! Incorrect login credentials. Please try again.",
        "NEWOR_USER_NOT_FOUND": "Oops! We can't find your credentials. Please try signing up first.",
      },
      "FIRST_NAME": "First name",
      "FIXED_DEPOSIT": "Fixed Deposit",
      "FORGOT_PASSWORD": "Forgot Password?",
      "FORGOT_PASSWORD_SUCCESS": "Password reset initiated!",
      "INITIAL_DEPOSIT": "Initial deposit",
      "INTEREST_RATE": "Rate of Interest",
      "LAST_NAME": "Last name",
      "LOGIN": "Login",
      "MATURITY_DATE": "Maturity date",
      "MOBILE_NUMBER": "Mobile number",
      "NEW_TO_NEWOR": "New to Newor?",
      "NO": "No",
      "PASSWORD": "Password",
      "PICTURE_UPDATED": "Picture updated successfully",
      "PLEASE_RESET_PASSWORD_EMAIL": "We have sent a mail to you to reset your password. Please update your password to login.",
      "PLEASE_VERIFY_EMAIL": "We have sent a link to your email for verification. Please complete the verification process to login.",
      "PRESENT_AMOUNT": "Amount present in the account",
      "PROFILE": "Profile",
      "PUBLIC_PROVIDENT_FUND": "Public Provident Fund",
      "RECURRING_DEPOSIT": "Recurring Deposit",
      "REMEMBER_CREDENTIALS": "Remember your credentials?",
      "RESET_PASSWORD": "Reset Password",
      "RESET_PASSWORD_SUCCESS": "You have successfully reset your password for your email. Please proceed with login",
      "SAME_INITIAL_AMOUNT": "Amount currently present in the account is the initial deposit?",
      "SAVINGS": "Savings Account",
      "SELECT_ASSET": "Select your Asset",
      "SELECT_DEPOSIT_TYPE": "Select Deposit type",
      "SELECT_START_AND_MATURITY_DATE": "Select Start date and Maturity date for {depositType}",
      "SELECT_START_DATE": "Select Start date for {depositType}",
      "SIGNUP": "Signup",
      "SIGNUP_SUCCESS": "Hurray! You have successfully signed up!",
      "START_DATE": "Start date",
      "SUBMIT": "Submit",
      "SUCCESS": "Success",
      "SURE_TO_LOGOUT": "Are you sure you want to logout of Newor?",
      "TOTAL_ASSET_VALUE": "Total Asset value",
      "WELCOME_TO_NEWOR": "Welcome to Newor!",
      "YES": "Yes",
    },
  },
}
`;

exports[`useTranslation should match snapshot for updated translation 1`] = `
LocalizedStrings {
  "LOGIN": "உள் நுழை",
  "_defaultLanguage": "ta",
  "_defaultLanguageFirstLevelKeys": Array [
    "LOGIN",
  ],
  "_interfaceLanguage": "en-US",
  "_language": "ta",
  "_opts": Object {
    "customLanguageInterface": [Function],
    "logsEnabled": true,
    "pseudo": false,
    "pseudoMultipleLanguages": false,
  },
  "_props": Object {
    "ta": Object {
      "LOGIN": "உள் நுழை",
    },
  },
}
`;
